File: src/main/java/com/fooddash/fooddash/service/ItemService.java
package com.fooddash.fooddash.service;

import com.fooddash.fooddash.dto.ItemRequest;
import com.fooddash.fooddash.entity.Item;
import com.fooddash.fooddash.entity.Vendor;
import com.fooddash.fooddash.repository.ItemRepository;
import com.fooddash.fooddash.repository.VendorRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class ItemService {

    @Autowired
    private ItemRepository itemRepository;

    @Autowired
    private VendorRepository vendorRepository;

    public List<Item> getAllItems() {
        return itemRepository.findByAvailable(true);
    }

    public List<Item> searchItems(String query) {
        return itemRepository.searchGlobal(query);
    }

    public List<Item> getItemsByCategory(String category) {
        return itemRepository.findByCategoryAndAvailable(category, true);
    }

    public List<String> getCategories() {
        return itemRepository.findDistinctCategories();
    }

    public List<Item> getVendorItems(Long vendorId) {
        return itemRepository.findByVendorId(vendorId);
    }

    public List<Item> getVendorItemsByUserId(Long userId) {
        Vendor vendor = vendorRepository.findByUserId(userId)
                .orElseThrow(() -> new RuntimeException("Vendor not found"));
        return itemRepository.findByVendorId(vendor.getId());
    }

    public Item addItem(Long userId, ItemRequest request) {
        Vendor vendor = vendorRepository.findByUserId(userId)
                .orElseThrow(() -> new RuntimeException("Vendor not found for user"));

        Item item = new Item();
        item.setVendor(vendor);
        item.setName(request.getName());
        item.setDescription(request.getDescription());
        item.setPrice(request.getPrice());
        item.setCategory(request.getCategory());
        item.setImageUrl(request.getImageUrl());
        item.setAvailable(request.getAvailable());
        item.setCalories(request.getCalories());
        item.setProtein(request.getProtein());
        item.setFat(request.getFat());
        item.setCarbs(request.getCarbs());

        return itemRepository.save(item);
    }

    public Item updateItem(Long itemId, ItemRequest request) {
        Item item = itemRepository.findById(itemId)
                .orElseThrow(() -> new RuntimeException("Item not found"));

        item.setName(request.getName());
        item.setDescription(request.getDescription());
        item.setPrice(request.getPrice());
        item.setCategory(request.getCategory());
        item.setImageUrl(request.getImageUrl());
        item.setAvailable(request.getAvailable());
        item.setCalories(request.getCalories());
        item.setProtein(request.getProtein());
        item.setFat(request.getFat());
        item.setCarbs(request.getCarbs());

        return itemRepository.save(item);
    }

    public void deleteItem(Long itemId) {
        itemRepository.deleteById(itemId);
    }
}

File: frontend/src/services/itemService.js
import api from './api';

const getAllItems = async (search = '', category = '') => {
    const params = new URLSearchParams();
    if (search) params.append('search', search);
    if (category) params.append('category', category);

    const response = await api.get(`/public/items?${params.toString()}`);
    return response.data;
};

const getCategories = async () => {
    const response = await api.get('/public/items/categories');
    return response.data;
};

const getVendorItemsPublic = async (vendorId) => {
    const response = await api.get(`/public/vendors/${vendorId}/items`);
    return response.data;
};

const getVendorItems = async () => {
    const response = await api.get('/vendor/items');
    return response.data;
};

const addItem = async (itemData) => {
    const response = await api.post('/vendor/items', itemData);
    return response.data;
};

const updateItem = async (itemId, itemData) => {
    const response = await api.put(`/vendor/items/${itemId}`, itemData);
    return response.data;
};

const deleteItem = async (itemId) => {
    const response = await api.delete(`/vendor/items/${itemId}`);
    return response.data;
};

const itemService = {
    getAllItems,
    getCategories,
    getVendorItemsPublic,
    getVendorItems,
    addItem,
    updateItem,
    deleteItem
};

export default itemService;
